### PR TITLE
Wrap icons when there is not enough space.

### PR DESCRIPTION
**Pull request by asvinb**
## Summary

Addresses issue #3839 

## Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

**Comment by asvinb** [comment]
@johnPhillips I know you worked on the IB, but can you check my notes in the "Relevant technical choices" of this PR? 
Thanks!

**Comment by johnPhillips** [comment]
Thanks @asvinb - you helped me a lot with the IB, so you are in a good position to evaluate the best approach here.

It sounds good to me, but I'll mention two things I noticed when sketching in the IB we arrived at.

1. When the idea content is verbose and the actions are *not* stacked, then the text has to wrap on hover, which is slightly janky:

Screenshot 2021-08-23 at 13 21 09

2. Although at 600 to 960px there does seem a lot of space, again bear in mind that there is no character limit on these ideas, so they can get quite long:

Screenshot 2021-08-23 at 13 20 33

So here we could have gone inline with the first two items, but the third might have looked a little strange.

In summary, I'd just suggest eyeballing your solution with some verbose ideas and seeing how it looks? Happy to go with whatever you decide 👍

**Comment by asvinb** [comment]
Thanks @johnPhillips . Based on our conversation, I've tweaked the PR so that items are stacked when the viewport is < 1280px.
@aaemnnosttv @felixarntz Is there a UX reason why we are showing the icons upon hover? I'am thinking whether it'll be best to show the icons across all breakpoints, especially we have #3907 where we are replacing icons by the loading spinner? 🤔

**Comment by felixarntz** [comment]
@asvinb 
> Is there a UX reason why we are showing the icons upon hover? I'am thinking whether it'll be best to show the icons across all breakpoints, especially we have #3907 where we are replacing icons by the loading spinner? 🤔 

I believe this was decided to not visually overload the user with too many action buttons. Any additional thoughts @marrrmarrr?

**Comment by johnPhillips** [comment]
If it is the case that we don't want to visually overload the UI with buttons, that's reasonable. But I would point out that we are showing them all on smaller viewports where there is less space: 

Screenshot 2021-08-24 at 10 23 58

This is my first issue with using `:hover` to hide the buttons. My second is that it increases complexity for us, especially as the buttons are also sometimes inline and sometimes wrapped. For example:
- How do we treat the [spinner behaviour ](https://github.com/google/site-kit-wp/issues/3907) on a button that is sometimes hidden unless your hover over it?
- What about verbose idea text that will have to wrap when you hover and expose the buttons (see my comment above)?

Just wondering if there is a more consistent way to keep the buttons visible but to lessen their visual impact / reduce the clutter. Maybe take a page out of WordPress's book? On the Posts page they use checkboxes and a single row of buttons that allows batch operations. They are also dealing with potentially verbose text, so they put all the actions underneath and only expose them on hover:

Screenshot 2021-08-24 at 12 13 03

Batch operations are obviously a fair amount of work. But if we always wrapped the buttons and made them visible, it could solve some of the issues I alluded to above. We could also look at secondary colours for some of the buttons to mute them slightly...

| Verbose content, small viewport | Large viewport |
| --- | --- |
| Screenshot 2021-08-24 at 12 27 21 | Screenshot 2021-08-24 at 12 28 14 |
 
I'm no UX expert and I'm certainly not a designer, so these might be terrible ideas 😂. 🤷‍♂️. All I'm saying is that perhaps we can find a better way to solve the clutter concern across all viewports?

**Comment by felixarntz** [comment]
@marrrmarrr @alex-Moulin Can you share your thoughts on @johnPhillips's feedback above? I tend to agree only showing the buttons on hover is causing issues here.

**Comment by alex-moulin** [comment]
@felixarntz @marrrmarrr @johnPhillips I would be keen to discuss this as I don't know for sure the reasons behind some of the decisions that were taken to design these elements. However, my 2 cents would be that instead of showing all the icons by default, we could potentially show 3 vertical dots on hover, and when interacted with, it would display the 3 options: Draft, Pin, Dismiss
This would allow us to keep the interface clean on all viewports but also could be a pattern we use throughout the plugin.

If possible, we could use an animation to show what's happening with the element, i.e. Dismiss would move the element to the left out of the list, Pin would move it to the right (as going into the other tab) and Draft would magnify it to take the users to the other screen.

**Comment by felixarntz** [comment]
Per https://github.com/google/site-kit-wp/issues/3839#issuecomment-910491145, we may actually want to circle back and go with the original PR approach here.

**Review by felixarntz — APPROVED** [review]
This actually looks good to me while testing! Paging @marrrmarrr for an additional approval.

**Comment by marrrmarrr** [comment]
Just tested it too, looks good!
